OSDOCS-5667: microshift release notes about OS and container runtime support (#58070)
openshift/openshift-docs: microshift_release_notes/microshift-4-13-release-notes.adoc

This release adds improvements related to the following components and concepts.

//L3 major categories with features in each as L4s, for example:
[id="microshift-4-13-rhel-edge"]

> === {op-system-base-full} {op-system-version}

Reviewer: ```suggestion
=== {op-system-ostree} {op-system-version-major}
```

//this is the plain RHEL attribute and version; do we need to create a RHEL for Edge attribute in the common_attributes or microshift_attributes files?

> * {product-title} runs on {op-system-base-full} version 9.2 or later.

Author: We need to include the minor version here because we do not support 9.0 or 9.1.

> * {product-title} runs on {op-system-base-full} version 9.2 or later.

Reviewer: Ack.

> * {product-title} uses crun and Control Group v2 (cgroupv2). {OCP} {ocp-version} defaults to Control Group v1. This difference should not result in any noticeable behavior differences for workloads, except in very unusual circumstances.

Reviewer: ```suggestion
* {product-title} uses crun and Control Group v2 (cgroupv2). {OCP} {ocp-version} defaults to Control Group v1. The divergence of control versions is not anticipated to have a noticeable behavior difference on workloads, except in very unusual circumstances.
```

This release adds improvements related to the following components and concepts.

//L3 major categories with features in each as L4s, for example:
[id="microshift-4-13-rhel-edge"]
=== {op-system-ostree} {op-system-version}
//this is the plain RHEL attribute and version; do we need to create a RHEL for Edge attribute in the common_attributes or microshift_attributes files?

* {product-title} runs on {op-system-ostree} version {op-system-version} or later.

// Some details of the next note are adapted from https://kubernetes.io/blog/2022/08/31/cgroupv2-ga-1-25/

> * {product-title} uses crun and Control Group v2 (cgroupv2). {OCP} {ocp-version} defaults to Control Group v1. The divergence of control group versions is not anticipated to have a noticeable behavior difference on most workloads. If workloads rely on the cgroup file system layout, they may need to be updated to be compatible with cgroupv2.

Reviewer 2: We should probably have a consistent naming scheme for cgroupv2. Above there is not a space between cgroup and v2

Author: Thanks, fixed.

. If you run third-party monitoring and security agents that depend on the cgroup file system, update the agents to versions that support cgroup v2.

> . If you run cAdvisor as a stand-alone DaemonSet for monitoring pods and containers, update it to v0.43.0 or later.

Reviewer 3: I think maybe we should flip the sentence to add the action first.
```suggestion
** Update to v0.43.0 or later if you are running cAdvisor as a standalone `DaemonSet` for monitoring pods and containers.
```
Standalone one word, see SSG https://redhat-documentation.github.io/supplementary-style-guide/#_s

Also, instead of a numbered list we should make these nested bullets with `**`

> . If you run cAdvisor as a stand-alone DaemonSet for monitoring pods and containers, update it to v0.43.0 or later.

Reviewer 3: Standalone one word, see SSG https://redhat-documentation.github.io/supplementary-style-guide/#_s

Also, instead of a numbered list we should make these nested bullets with **

> . If you deploy Java applications with the JDK, ensure you are using JDK 11.0.16 and later or JDK 15 and later, which fully support cgroup v2.

Reviewer 3: Should this be OpenJDK?

> . If you deploy Java applications with the JDK, ensure you are using JDK 11.0.16 and later or JDK 15 and later, which fully support cgroup v2.

Reviewer 4: There are various stream maintained. 11 is a LTS so if someone is on that stream it needs to be 11.0.16. Else any JDK 15+ shoud work. i.e. 16, 17.. through the just released 20.
